### PR TITLE
chore: add temporary env flag for displaying the alteration UI

### PR DESCRIPTION
**Pull request by EmiliaMakelaVincit**
As the alteration reporting feature is not yet completed on the handler side for some time, conditionally hiding it will allow us to push other changes to production in the meantime.

A similar flag was also added for the handler side for use when the development begins on that side.

## Testing :alembic:
The alteration report features can be turned on by setting the environment variable added here (`NEXT_PUBLIC_ENABLE_ALTERATION_FEATURES` to `true`). If the environment variable is missing (i.e. on production) or set to any other value, the features are hidden from the end user.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2893.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2893.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2893.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2893.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/applicant/src/components/applications/pageContent/DecisionSummary.tsx [hunk 1]
```diff
@@ -99,31 +99,35 @@ const DecisionSummary = ({ application }: Props): JSX.Element => {
...
-                router.push(
-                  `${ROUTES.APPLICATION_ALTERATION}?id=${application.id}`
-                )
-              }
-            >
-              {t('common:applications.decision.actions.reportAlteration')}
-            </Button>
-          )}
-      </$ActionContainer>
+      {process.env.NEXT_PUBLIC_ENABLE_ALTERATION_FEATURES === 'true' && (
```
Comment (sirtawast) [inline frontend/benefit/applicant/src/components/applications/pageContent/DecisionSummary.tsx:102]: btw, we have this handy utility function: 

https://github.com/City-of-Helsinki/yjdh/blob/ebbe9e5fd2cb21ec457845e5fe16cdfa617e655a/frontend/benefit/shared/src/utils/common.ts#L29

**Review by sirtawast — APPROVED** [review]

File: frontend/benefit/applicant/src/components/applications/pageContent/DecisionSummary.tsx (hunk 1 above)
Comment (sirtawast) [reply frontend/benefit/applicant/src/components/applications/pageContent/DecisionSummary.tsx:102]: I wonder why it doesn't include `(int)1`

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2893.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2893.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2893.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2893.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2893.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**